lib/parallel_tests/gherkin/runner.rb

```diff
@@ -67,11 +70,25 @@ def summarize_results(results)
         end
 
         def cucumber_opts(given)
-          if given =~ (/--profile/) || given =~ (/(^|\s)-p /)
-            given
-          else
-            [given, profile_from_config].compact.join(" ")
+          initial =
+            if given =~ (/--profile/) || given =~ (/(^|\s)-p /)
+              given
+            else
+              [given, profile_from_config].compact.join(" ")
+            end
+
+          opts_as_individuals = initial.scan(/\S+\s\S+/)
```

Comment:
we now split on each space separated opt. So here for example `-o "-p foo -p bar -t @footag -t @bartag --retry 3"` would become an array of those 5 items.

Reply:
I think you want `string.shellsplit` from `require 'shellwords'` package